### ISSUE TITLE
Jena interop

### ISSUE THREAD

**Issue by umbreak** [filip26/titanium-json-ld#77]
It would be great to implement some conversion from the RdfDatasets of Jena and the ones from this library.

I know this should be implemented in another repo, so this library does not directly depend on Jena.

For your information, I've submitted an issue on the Jena side, and that's their response: https://issues.apache.org/jira/browse/JENA-1948

**Comment by jmvanel** [comment]
I did this:
```java
  /** populate a Jena Dataset from a Titanium Dataset */
  public static DatasetGraph populateDataset( RdfDataset titaniumIn, DatasetGraph jenaToUpdate) {
```
https://github.com/jmvanel/semantic_forms/blob/master/scala/jsonld/src/main/java/titaniumJena/Titanium2Jena.java

Andy and Filip know about it .
Bug reports wellcome.

**Comment by umbreak** [comment]
I guess to provide conversions in both directions are a pre-requisite for Jena in order for them to even considering adding Titanium for JSON-LD-11.

Great work!

**Comment by filip26** [comment]
Hi,
thank you @umbreak and @jmvanel . Jena integration is a very cool idea!

I can provide all needed support, cooperation, but I'm not an expert on Jena, and I would very much appreciate if someone else could take care of the integration - e.g. a list of what needs to be done in order to proceed would be very helpful.

Please let me know what you think.

**Comment by jmvanel** [comment]
I just wrote this to populate a Titanium Dataset from a Jena Graph
https://github.com/jmvanel/semantic_forms/blob/master/scala/jsonld/src/main/java/titaniumJena/Jena2Titanium.java

**Comment by afs** [comment]
There has been some progress on a very basic Jena reader https://issues.apache.org/jira/browse/JENA-1948. Functionally, it seems to work.

There have been users working with data such as [Web of Things (WoT) Thing Description 1.1](https://www.w3.org/TR/wot-thing-description11/) which is JSON-LD 1.1.

Adding a module to Titanium is one way to go but maybe not the best long-term. Learning from other adapters (CommonsRDF, jsonld-java), the problem is that it imposes a cost on the project to track Jena versions, even if just for version number (and many users do follow version numbers and do not wish to self-assemble variations). Nowadays, with dependabot, version-tracking updates Jena dependencies happens quite promptly.

**Comment by afs** [comment]
Apache Jena 4.2.0 has been released with support for reading JSON-LD 1.1 using titanium-json-ld.

[Announcement of Jena 4.2.0](https://lists.apache.org/thread.html/r90f6a4ceed08cf67e6d13ab84e5683f7a2fd5d96c3d4591adea8e06f%40%3Cusers.jena.apache.org%3E).

[Documentation about JSON-LD 1.1 support](https://jena.apache.org/documentation/io/json-ld-11.html) in this release.

Thank you very much for this project!

**Comment by filip26** [comment]
Hi Andy,
that's great news. Thank you for integrating this library with Jena!
